merkle.go

```diff
@@ -0,0 +1,189 @@
+package fastmerkle
+
+import (
+	"errors"
+	"fmt"
+	"runtime"
+)
+
+var (
+	errEmptyDataSet = errors.New("empty data set provided")
+)
+
+// GenerateMerkleTree generates a Merkle tree based on the input data
+func GenerateMerkleTree(inputData [][]byte) (*MerkleTree, error) {
+	// Check if the data set is valid
+	if !isValidDataSet(inputData) {
+		return nil, errEmptyDataSet
+	}
```

Comment:
I think no need to create function because it's simple check. And also you can just check `len` because `len` returns 0 in case of nil

Reply:
You're right 💯 

Simplified:

[1e8585b](https://github.com/Trapesys/fastmerkle/pull/1/commits/1e8585b9beb318ccacd5bfc3a23f6782cf49d217)